### PR TITLE
GEODE-10259: Update protocol ordinal to 1.14.0 (125)

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**

### CODE REVIEW

File: cppcache/src/ClientProxyMembershipID.hpp [hunk 1]
```diff
@@ -51,26 +46,37 @@ class ClientProxyMembershipID : public DSMemberForVersionStamp {
...
   static void increaseSynchCounter();
+
   static std::shared_ptr<Serializable> createDeserializable() {
     return std::make_shared<ClientProxyMembershipID>();
   }
-  // Do an empty check on the returned value. Only use after handshake is done.
+
+  /**
+   * Do an empty check on the returned value. Only use after handshake is done.
+   */
```
Comment (pdxcodemonkey) [inline cppcache/src/ClientProxyMembershipID.hpp:69]: I'm not sure this comment adds any value.  Also the name of this function doesn't appear to describe it correctly - it returns member var `clientId_`, maybe just call it getClientId?

**Review by pdxcodemonkey — CHANGES_REQUESTED** [review]
This is huge - thanks!  Just a couple of questions and some suggestions around naming...

File: cppcache/integration/test/PdxJsonTypeTest.cpp [hunk 2]
```diff
@@ -101,8 +101,7 @@ TEST(PdxJsonTypeTest, testGfshQueryJsonInstances) {
               cache.createPdxInstanceFactory(gemfireJsonClassName)
                   .writeString("entryName", "java-domain-class-entry")
                   .create());
-  ASSERT_THROW(execution.withArgs(query).execute("QueryFunction"),
-               CacheServerException);
+  ASSERT_NO_THROW(execution.withArgs(query).execute("QueryFunction"));
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/integration/test/PdxJsonTypeTest.cpp:104]: I don't know. The original author left no clue as to why and if that was truly the wanted behavior. The test seems to test a few things in a single test. The name of the test leaves little clue. I don't know why the original was even throwing an exception here.

File: cppcache/src/ClientHealthStats.cpp [hunk 3]
```diff
@@ -14,62 +14,58 @@
...
-  output.writeInt(static_cast<int64_t>(m_processCpuTime));
-  m_updateTime->toData(output);
+  output.writeInt(static_cast<int64_t>(gets_));
+  output.writeInt(static_cast<int64_t>(puts_));
+  output.writeInt(static_cast<int64_t>(misses_));
+  output.writeInt(static_cast<int32_t>(cacheListenerCallsCompleted_));
+  output.writeInt(static_cast<int32_t>(threads_));
+  output.writeInt(static_cast<int32_t>(cpus_));
+  output.writeInt(static_cast<int64_t>(processCpuTime_));
+  updateTime_->toData(output);
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/ClientHealthStats.cpp:36]: Not sure what you mean by "version of the stat file"? The field type is recorded in the stat file so the size change has no real affect on the consumer of the file. The field sizes change in the protocol some time ago when we dropped support for `int` type stats.

File: cppcache/src/ClientProxyMembershipID.cpp [hunk 4]
```diff
@@ -31,19 +26,20 @@
 #include "Version.hpp"
 #include "util/Log.hpp"
 
-#define DCPORT 12334
-#define VMKIND 13
-#define ROLEARRLENGTH 0
+namespace {
+constexpr int32_t kVersionMask = 0x8;
+constexpr int8_t kVmKindLoaner = 13;
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/ClientProxyMembershipID.cpp:31]: LOL! Probably auto correct.

File: cppcache/src/ClientProxyMembershipID.cpp [hunk 5]
```diff
@@ -247,31 +243,26 @@ Serializable* ClientProxyMembershipID::readEssentialData(DataInput& input) {
...
-  return this;
 }
 
 void ClientProxyMembershipID::readAdditionalData(DataInput& input) {
   // Skip unused UUID (16) and weight (0);
   input.advanceCursor(17);
 }
 
-void ClientProxyMembershipID::increaseSynchCounter() { ++synch_counter; }
+void ClientProxyMembershipID::increaseSynchCounter() { ++synchCounter; }
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/ClientProxyMembershipID.cpp:265]: It bugs you too... I thought about it...

File: cppcache/src/ClientProxyMembershipID.hpp [hunk 6]
```diff
@@ -36,10 +35,6 @@ namespace apache {
 namespace geode {
 namespace client {
 
-using internal::DSFid;
-
-class ClientProxyMembershipID;
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/ClientProxyMembershipID.hpp:41]: I know, right!

File: cppcache/src/ClientProxyMembershipID.hpp (hunk 1 above)
Comment (pivotal-jbarrett) [reply cppcache/src/ClientProxyMembershipID.hpp:69]: I didn't want to get that deep into refactoring this. I wanted to keep to protocol. I just hate seeing documentation comments using the wrong comment format. We could wack the whole comment. I will leave the refactoring of the name and purpose up to someone else.

File: cppcache/src/ClientProxyMembershipID.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/ClientProxyMembershipID.cpp:265]: You should take a look at that value. I am pretty sure that is always 3. Not sure what the point is.

File: cppcache/src/ClientProxyMembershipID.hpp (hunk 1 above)
Comment (pivotal-jbarrett) [reply cppcache/src/ClientProxyMembershipID.hpp:69]: Ok, I am going to fix it. I didn't even look at the implementation previously. You are right, it is just a getter.

File: cppcache/src/ClientProxyMembershipID.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (pdxcodemonkey) [reply cppcache/src/ClientProxyMembershipID.cpp]: Seriously - what the heck does this thing do??????  It starts at 2, and `increaseSyncCounter` is called by the ThinClientPoolDM ctor, so is there only ever one of these(???).  If so, it's yeah just always 3.  It gets written into both clientId and memberId, but I also have no idea what, if any, information in those things is ever unpacked on the server side.

**Review by pdxcodemonkey — APPROVED** [review]

**Pull request merged**